Move trackset-features into library subfolder

xeruf opened the pull request:
Follow-up to my previous PR. The second commit is a pure refactor/reformat, no code changes. Tested with SCons & CMake.

xeruf committed:
Move childModel to BaseTrackSetFeature

xeruf committed:
Move all trackset-features into library subdirectory

xerus2000 commented:
Follow-up to my previous PR. The second commit is a pure refactor/reformat, no code changes. Tested with SCons & CMake.

xerus2000 commented:
@Be-ing can't request review directly :)

There is the "Triage" role on GitHub now, which would give access to such things, are you using that?

uklotzde commented on src/library/trackset/crate/cratetablemodel.cpp:
```diff
@@ -103,17 +102,7 @@ bool CrateTableModel::addTrack(const QModelIndex& index, QString location) {
-            | TRACKMODELCAPS_LOADTOPREVIEWDECK
-            | TRACKMODELCAPS_REMOVE_CRATE
-            | TRACKMODELCAPS_RESETPLAYED;
+    CapabilitiesFlags caps = TRACKMODELCAPS_NONE | TRACKMODELCAPS_RECEIVEDROPS | TRACKMODELCAPS_ADDTOPLAYLIST | TRACKMODELCAPS_ADDTOCRATE | TRACKMODELCAPS_ADDTOAUTODJ | TRACKMODELCAPS_EDITMETADATA | TRACKMODELCAPS_LOADTODECK | TRACKMODELCAPS_LOADTOSAMPLER | TRACKMODELCAPS_LOADTOPREVIEWDECK | TRACKMODELCAPS_REMOVE_CRATE | TRACKMODELCAPS_RESETPLAYED;
```
Unfortunately clang-format only doesn't recognize this variant. The operands are preserved on separate lines when placing the `|` operator *after* each operand.

This code has been written before the introduction of clang-format.

xerus2000 replied:
Fixed :+1:

uklotzde reviewed (commented):
Good idea to file a PR before actually changing any code.

Please check the formatting of each file, especially check for long lines. Sometimes you need need to give clang-format a hint to get it right.

xeruf committed:
Clean up CapabilitiesFlags and apply clang format

xerus2000 commented on src/library/banshee/bansheeplaylistmodel.cpp:
```diff
@@ -75,71 +75,75 @@ void BansheePlaylistModel::setTableModel(int playlistId) {
-                 CLM_COMPOSER " TEXT, "
-                 CLM_PREVIEW " TEXT)");
+        QString strQuery(
+                "CREATE TEMP TABLE IF NOT EXISTS %1"
```
I had to write string-variable instead of variable-string for clang-format, in case anyone wants to ask. Suggestions for doing it better are welcome ^^

Holzhaus replied:
`QStringLiteral("CREATE TEMP TABLE IF NOT EXISTS ") + m_tempTableName + QStringLiteral( [...] )`

xerus2000 replied:
Like this? bbabfad11c

Holzhaus replied:
I think you can put string literals and preprocessor string literals into the same qstringliteral (without + in between).

xerus2000 replied:
Which will probably make clang-format reformat it to mess again, that is why I asked in the first place ^^

xerus2000 replied:
Is there any functional advantage of using QStringLiteral here?

Holzhaus replied:
Yes: https://woboq.com/blog/qstringliteral.html

xerus2000 commented on src/library/baseexternalplaylistmodel.cpp:
```diff
@@ -114,23 +105,22 @@ void BaseExternalPlaylistModel::setPlaylist(QString playlist_path) {
-                 QString::number(playlistId));
+            "CREATE TEMPORARY VIEW IF NOT EXISTS %1 AS "
+            "SELECT %2 FROM %3 WHERE playlist_id = %4")
+                                  .arg(f.escapeString(playlistViewTable),
```
I'm unsure if this is ho it should look like

Holzhaus replied:
Use QStringLiteral to make use of the String builder. And use QSqlQuery::bindValue for the playlist id. It's unlikely that an SQL injection is possible since QString::number is used, but better safe than sorry. And this also prevents people from doing dangerous things when they touch these lines in the future.

xerus2000 replied:
Hm, I'd prefer not changing any function calls since I am not really familiar with this code.

Holzhaus replied:
It's not complicated, just replace `%4` with `:playlist_id` and run `query.bindValue(":playlist_id", playlistId)` after `query.prepare(queryString);`

xerus2000 replied:
Like this? bbabfad11c

xerus2000 commented on src/library/hiddentablemodel.cpp:
```diff
@@ -22,13 +21,16 @@ void HiddenTableModel::setTableModel(int id) {
+    query.prepare(
+            "CREATE TEMPORARY VIEW IF NOT EXISTS " + tableName +
+            " AS "
+            "SELECT " +
```
This looks horrible imo, if someone has a way to write it that fits clang-format, you are welcome!

daschuer replied:
The nice thing with clang-format is that we can stop arguing about horrible or nice. 

It contains the common denominator that works for all. Of case you can find places where clang-format "fails" but that is OK by definition.

You may try to trick it but that is absolutely not necessary. Just go ahead.

xerus2000 replied:
Inspired by @Holzhaus I changed it to this, does that make sense? 63bf05a

Holzhaus replied:
AFAIK bindValue only works for values, not table names, column lists or boolean expressions.

xeruf committed:
Use QStringLiteral and value binding for library queries

The trailing slashes prevent clang-format from wrapping that text into a big chunk, see:
https://stackoverflow.com/a/34362422

daschuer commented on src/library/trackset/baseplaylistfeature.h:
```diff
@@ -79,22 +80,21 @@ class BasePlaylistFeature : public BaseTrackSetFeature {
-    QAction* m_pExportTrackFilesAction;
-    QAction* m_pDuplicatePlaylistAction;
-    QAction* m_pAnalyzePlaylistAction;
+    QAction* m_pCreatePlaylistAction{};
```
We don't initialize variables in the header file.
This is unexpected in the Mixxx code base and  introduces a static hidden constructor, we like to avoid.

daschuer replied:
If you like you may use parented_ptr to document the ownership.

xerus2000 replied:
oh, I think that was an accident ^^

daschuer commented on src/library/trackset/crate/cratetablemodel.cpp:
```diff
@@ -37,23 +36,23 @@ void CrateTableModel::selectCrate(CrateId crateId) {
+            "SELECT %2 FROM %3 "
+            "WHERE %4 IN (%5) "
+            "AND %6=0")
+                                  .arg(tableName,
```
Was this done by clang-format? I think we have improved it since than.
I would expect that this has no or 8 space indent to the line above.
like that: 
```
    QString queryString =
            QString("CREATE TEMPORARY VIEW IF NOT EXISTS %1 AS "
                    "SELECT %2 FROM %3 "
                    "WHERE %4 IN (%5) "
                    "AND %6=0")
                    .arg(tableName,
                            columns.join(","),
                            LIBRARY_TABLE,
                            LIBRARYTABLE_ID,
                            CrateStorage::formatSubselectQueryForCrateTrackIds(
                                    crateId),
                            LIBRARYTABLE_MIXXXDELETED);
```

daschuer replied:
The are as some more occurrences in this PR. 

Since this is a code styling PR, I suggest to rebase -i and amend all commits using our pre-commit hock one by one.

xerus2000 replied:
I was questioning this result as well, see https://github.com/mixxxdj/mixxx/pull/2715#discussion_r415294841 ;)

xerus2000 replied:
Ah, putting the "QStringLiteral" in a new line fixes it, since the continuation indent is relative to the original ;)

xeruf committed:
Clean up HiddenTableModel::setTableModel

xerus2000 commented:
I have rebased on master and presumably fixed all mentioned problems :)

Holzhaus commented on src/library/trackmodel.h:
```diff
@@ -29,26 +29,27 @@ class TrackModel {
+        TRACKMODELCAPS_UNHIDE            = 0x08000u,
+        TRACKMODELCAPS_PURGE             = 0x10000u,
+        TRACKMODELCAPS_REMOVE_PLAYLIST   = 0x20000u,
+        TRACKMODELCAPS_REMOVE_CRATE      = 0x40000u,
```
Can we make this an enum class and use QFlags for this? (We can also do this in another PR)

Also, for bitfields I usually prefer this notation over hex, because it becomes much easier to spot mistakes:

    0
    1
    1 << 1
    1 << 2
    1 << 3

xerus2000 replied:
Yes we can do that, but please hold on with such functional changes - this is supposed to be a pure refactoring PR. The more we change, the larger the chance for conflicts with #2717, and I don't fancy that.

Holzhaus replied:
okay.

xerus2000 replied:
Fixed in #2810 :)

Holzhaus commented on src/library/trackset/crate/cratestorage.cpp:
```diff
@@ -24,44 +20,41 @@ const QString CRATESUMMARY_TRACK_COUNT = "track_count";
-                CRATETRACKSTABLE_TRACKID,
-                LIBRARY_TABLE,
-                LIBRARYTABLE_ID);
+        "LEFT JOIN %3 ON %3.%4=%1.%2")
+                                         .arg(
+                                                 CRATE_TABLE,
+                                                 CRATETABLE_ID,
+                                                 CRATE_TRACKS_TABLE,
+                                                 CRATETRACKSTABLE_CRATEID);
```
Is that how our pre-commit hooks formatted it? that looks strange.

xerus2000 replied:
Yeah, I didn't check that, see https://github.com/mixxxdj/mixxx/pull/2715#discussion_r427163358 - fixing it rn

Holzhaus commented on src/library/trackset/crate/cratestorage.cpp:
```diff
@@ -24,44 +20,41 @@ const QString CRATESUMMARY_TRACK_COUNT = "track_count";
+                                                 CRATE_TRACKS_TABLE,
+                                                 CRATETRACKSTABLE_CRATEID);
+
+const QString kLibraryTracksJoin = kCrateTracksJoin + QString(" LEFT JOIN %3 ON %3.%4=%1.%2").arg(CRATE_TRACKS_TABLE, CRATETRACKSTABLE_TRACKID, LIBRARY_TABLE, LIBRARYTABLE_ID);
```
long line

Holzhaus commented on src/library/trackset/crate/cratestorage.cpp:
```diff
@@ -24,44 +20,41 @@ const QString CRATESUMMARY_TRACK_COUNT = "track_count";
-                LIBRARYTABLE_MIXXXDELETED,
-                CRATESUMMARY_TRACK_COUNT,
-                CRATESUMMARY_TRACK_DURATION);
+        "COUNT(CASE %2.%4 WHEN 0 THEN 1 ELSE NULL END) AS %5,"
+        "SUM(CASE %2.%4 WHEN 0 THEN %2.%3 ELSE 0 END) AS %6 "
+        "FROM %1")
+                                                .arg(
+                                                        CRATE_TABLE,
+                                                        LIBRARY_TABLE,
+                                                        LIBRARYTABLE_DURATION,
+                                                        LIBRARYTABLE_MIXXXDELETED,
+                                                        CRATESUMMARY_TRACK_COUNT,
+                                                        CRATESUMMARY_TRACK_DURATION);
```
this also looks strange

xerus2000 replied:
wait, you can comment on multiple lines? This is amazing, didn't know that :o

Holzhaus commented on src/library/trackset/crate/cratestorage.cpp:
```diff
@@ -440,49 +388,26 @@ uint CrateStorage::countCrateTracks(CrateId crateId) const {
-            CRATETABLE_ID,
-            CRATETABLE_NAME,
-            escapedCrateNameLike);
+    return QString("SELECT DISTINCT %1 FROM %2 JOIN %3 ON %4=%5 WHERE %6 LIKE %7 ORDER BY %1").arg(CRATETRACKSTABLE_TRACKID, CRATE_TRACKS_TABLE, CRATE_TABLE, CRATETRACKSTABLE_CRATEID, CRATETABLE_ID, CRATETABLE_NAME, escapedCrateNameLike);
```
Break this long line (and all others)

Holzhaus commented on src/library/trackset/crate/cratetablemodel.cpp:
```diff
@@ -124,6 +124,7 @@ TrackModel::CapabilitiesFlags CrateTableModel::getCapabilities() const {
             qWarning() << "Failed to read create" << m_selectedCrate;
         }
     }
+    
```
trailing whitespace

xerus2000 replied:
I think that was intentional to separate the different parts of the function

Holzhaus replied:
A blank line is okay, but not trailing whitespace (e.g. spaces)

xerus2000 replied:
Ah, didn't see that in the diff, will fix.

xeruf committed:
Fix formatting in CreateFeature classes

xerus2000 commented:
I would like to have this PR merged as there don't seem to any substantial issues left. Deferred tasks:
- [ ] TrackModel Capabilities to enum: https://github.com/mixxxdj/mixxx/pull/2715#discussion_r427221201

@Holzhaus please check if I resolved https://github.com/mixxxdj/mixxx/pull/2715#discussion_r415307496 correctly

xeruf committed:
Inline anonymous method in CrateFeature

Holzhaus commented on src/library/playlisttablemodel.cpp:
```diff
@@ -100,8 +100,7 @@ void PlaylistTableModel::setTableModel(int playlistId) {
     columns[4] = LIBRARYTABLE_COVERART;
-    setTable(playlistTableName, LIBRARYTABLE_ID, columns,
-            m_pTrackCollectionManager->internalCollection()->getTrackSource());
+    setTable(playlistTableName, LIBRARYTABLE_ID, columns, m_pTrackCollectionManager->internalCollection()->getTrackSource());
```
long line

xerus2000 replied:
https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/About.20coding.20style/near/198301456

Holzhaus commented on src/library/trackset/crate/cratestorage.cpp:
```diff
@@ -529,15 +473,9 @@ CrateSummarySelectResult CrateStorage::selectCratesWithTrackCount(const QList<Tr
-                    CRATETABLE_NAME));
+    FwdSqlQuery query(m_database,
+            QString("SELECT %1,%2 FROM %3 JOIN %4 ON %5 = %6 WHERE %7 LIKE :crateNameLike ORDER BY %1")
+                    .arg(CRATETRACKSTABLE_TRACKID, CRATETRACKSTABLE_CRATEID, CRATE_TRACKS_TABLE, CRATE_TABLE, CRATETABLE_ID, CRATETRACKSTABLE_CRATEID, CRATETABLE_NAME));
```
long line

Holzhaus commented on src/library/playlisttablemodel.cpp:
```diff
@@ -272,24 +270,24 @@ bool PlaylistTableModel::isColumnHiddenByDefault(int column) {
         caps |= TRACKMODELCAPS_REMOVE;
     }
-    if (m_pTrackCollectionManager->internalCollection()->getPlaylistDAO().getHiddenType(m_iPlaylistId)== PlaylistDAO::PLHT_SET_LOG) {
+    if (m_pTrackCollectionManager->internalCollection()->getPlaylistDAO().getHiddenType(m_iPlaylistId) == PlaylistDAO::PLHT_SET_LOG) {
```
long line

xerus2000 replied:
A line break here looks a little weird:
![image](https://user-images.githubusercontent.com/13354331/82544885-1892fb80-9b56-11ea-87b4-2f17953fccd8.png)

xerus2000 replied:
But I fixed it in https://github.com/mixxxdj/mixxx/pull/2715/commits/bd4a67d8007602eccfc0b224d4ad94b99a6abb0b#diff-f7dd1ed098b395653d5cdc492c7f7b20 :)

Holzhaus commented on src/library/librarytablemodel.cpp:
```diff
@@ -50,8 +53,7 @@ void LibraryTableModel::setTableModel(int id) {
     tableColumns << LIBRARYTABLE_COVERART;
-    setTable(tableName, LIBRARYTABLE_ID, tableColumns,
-             m_pTrackCollectionManager->internalCollection()->getTrackSource());
+    setTable(tableName, LIBRARYTABLE_ID, tableColumns, m_pTrackCollectionManager->internalCollection()->getTrackSource());
```
long line

Holzhaus commented on .clang-format:
```diff
@@ -18,6 +18,7 @@ AlignOperands: false
 AllowShortFunctionsOnASingleLine: None
 AllowShortIfStatementsOnASingleLine: false
 AllowShortLoopsOnASingleLine: false
+AlwaysBreakBeforeMultilineStrings: true
```
Why did you change this?

uklotzde replied:
@xerus2000 Please do NOT change those settings without any notice!!

xerus2000 replied:
oh I'm sorry, I changed that for testing and forgot it

xerus2000 replied:
Nonetheless, I would actually propose keeping it, it turns this:
![image](https://user-images.githubusercontent.com/13354331/82541361-a0760700-9b50-11ea-96d2-c05e6c590341.png)
into this:
![image](https://user-images.githubusercontent.com/13354331/82541382-aa980580-9b50-11ea-9baf-21072f37ffee.png)
which is much more readable imho

Holzhaus replied:
I don't really care and think both are okay. 

@mixxxdj/developers opinions?

xerus2000 replied:
@Holzhaus I reverted the change for now so we can merge :)

xerus2000 replied:
Discussion moved: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/About.20coding.20style/near/198313697

Holzhaus commented on src/library/hiddentablemodel.cpp:
```diff
@@ -1,34 +1,30 @@
     QStringList columns;
     columns << "library." + LIBRARYTABLE_ID;
-    QString filter("mixxx_deleted=1");
+    // clang-format off
```
Is this really necessary? How does it look without it?

xerus2000 replied:
It breaks the line after every single "+", which makes it horribly unreadable

xerus2000 replied:
Nevermind, found a way to work around it.

Holzhaus reviewed (commented):
Thanks, looks mostly good.

xeruf committed:
Reformat long lines & clean up tablemodels

xerus2000 commented:
@Holzhaus all should be fixed now - again, please check above (https://github.com/mixxxdj/mixxx/pull/2715#discussion_r415307496) if I did it correctly and resolve the comment

Holzhaus commented:
OK, thanks. But please don't force push after review has started unless it's to fix non-building commits that prevent us from bisecting later.
Btw, there are a bunch of merge conflicts now.

xerus2000 commented:
> OK, thanks. But please don't force push after review has started unless it's to fix non-building commits that prevent us from bisecting later.

I wanted to get that clang-format change out of the history because it should've never been there and might cause unnecessary confusion. If I shouldn't do that, sure, I'll simply push a commit next time.

> Btw, there are a bunch of merge conflicts now.

Yep, that was to be expected, that's why I wanted to merge one of my PRs ASAP ^^

xeruf committed:
Merge branch 'master' into refactor-basetrackset

xeruf committed:
Move TrackSetTableModel into trackset folder

xerus2000 commented:
@Holzhaus all resolved :)

xeruf committed:
Use QStringLiteral for BansheePlaylistModel queries

xerus2000 commented:
Sorry for the other force-pushes, they only concerned recent commits :)

uklotzde commented on src/library/trackset/crate/cratetablemodel.cpp:
```diff
@@ -74,20 +76,20 @@ bool CrateTableModel::addTrack(const QModelIndex& index, QString location) {
-                << "Failed to add track"
-                << location
-                << "to library";
+                 << "Failed to add track"
```
These indentation changes are inconsistent with our .clang-form. Applies to multiple occurrences in this PR!

xerus2000 replied:
Nope, I have applied clang-format to everything! 
I double-checked for this file, re-running clang-format changes nothing.

uklotzde replied:
Double-checked again and confirmed. Strange, I remember that this was indented with 8 spaces instead of aligning the `<<` operators on subsequent lines.

xerus2000 commented:
@Holzhaus @daschuer @uklotzde can you give green light?

xeruf committed:
Replace empty destructors with default

uklotzde commented on src/library/banshee/bansheeplaylistmodel.cpp:
```diff
@@ -75,73 +75,74 @@ void BansheePlaylistModel::setTableModel(int playlistId) {
-                 CLM_PREVIEW " TEXT)");
-        if (!query.exec(strQuery.arg(m_tempTableName))) {
+        if (!query.exec(QStringLiteral(
+                    "CREATE TEMP TABLE IF NOT EXISTS %1 (" //
```
Please remove these empty end-of-line comments that don't belong to the code and don't add any value.

xerus2000 replied:
I don't know how else to get to a satisfactory result in this code, this is what the auto-formatter does otherwise:
![image](https://user-images.githubusercontent.com/13354331/82602352-3e99b980-9bb1-11ea-8ba2-13339012ab93.png)
This is the only alternative I found, but it is semantically very wrong because the type of one column is in the same line with the name of the next:
![image](https://user-images.githubusercontent.com/13354331/82602472-7143b200-9bb1-11ea-92f6-8d1164471682.png)

xerus2000 replied:
I got to a satisfactory solution for the part below (https://github.com/mixxxdj/mixxx/pull/2715/commits/bc862eda302bbded3d4790c315db6970d7ec83a4), but this doesn't work here because there's a different type for each column.

Holzhaus replied:
Are you using pre-commit? If you do you shouldn't get a single long line thanks to the line-length hook.

xerus2000 replied:
I am using pre-commit now. Still ending up with that long line ^^

xerus2000 replied:
@daschuer

uklotzde reviewed (changes requested):
Please remove all empty end-of-line comments and let the clang-format do its work.

The remaining changes seem to be a straight-forward refactoring and reformatting. I haven't checked every single line and trust you that this is the case.

xeruf committed:
Replace weirdly formatted QStringLiteral by joined QStringList

daschuer commented on src/library/browse/browsetablemodel.cpp:
```diff
@@ -160,10 +160,9 @@ TrackPointer BrowseTableModel::getTrack(const QModelIndex& index) const {
-            + "\n" + trackRef.getLocation());
+                0,
+                tr("Mixxx Library"),
+                tr("Could not load the following file because it is in use by Mixxx or another application.") + "\n" + trackRef.getLocation());
```
Long line, requires Github vertical scrolling. Why does clang-format not kick in here?

xerus2000 replied:
clang-format does kick in - if you look above it actually put all that stuff into one line in the first place. Maybe you need to tweak the penalty for long lines or smth like that.

daschuer commented on src/library/hiddentablemodel.cpp:
```diff
@@ -40,8 +38,7 @@ void HiddenTableModel::setTableModel(int id) {
     tableColumns << LIBRARYTABLE_ID;
-    setTable(tableName, LIBRARYTABLE_ID, tableColumns,
-             m_pTrackCollectionManager->internalCollection()->getTrackSource());
+    setTable(tableName, LIBRARYTABLE_ID, tableColumns, m_pTrackCollectionManager->internalCollection()->getTrackSource());
```
Long Line.

daschuer commented on src/library/library.cpp:
```diff
@@ -127,37 +117,33 @@ Library::Library(
-            m_pAnalysisFeature, &AnalysisFeature::analyzeTracks);
-    connect(m_pCrateFeature, &CrateFeature::analyzeTracks,
-            m_pAnalysisFeature, &AnalysisFeature::analyzeTracks);
+    connect(m_pPlaylistFeature, &PlaylistFeature::analyzeTracks, m_pAnalysisFeature, &AnalysisFeature::analyzeTracks);
```
long line.

daschuer commented on src/library/library.cpp:
```diff
@@ -127,37 +117,33 @@ Library::Library(
-            this, &Library::onPlayerManagerTrackAnalyzerProgress);
-    connect(pPlayerManager, &PlayerManager::trackAnalyzerIdle,
-            this, &Library::onPlayerManagerTrackAnalyzerIdle);
+    connect(pPlayerManager, &PlayerManager::trackAnalyzerProgress, this, &Library::onPlayerManagerTrackAnalyzerProgress);
```
long line.

daschuer commented on src/library/library.cpp:
```diff
@@ -463,10 +448,9 @@ void Library::slotRequestAddDir(QString dir) {
-                tr("Could not add the directory to your library. Either this "
-                    "directory is already in your library or you are currently "
-                    "rescanning your library."));
+        QMessageBox::information(0, tr("Add Directory to Library"), tr("Could not add the directory to your library. Either this "
```
long line.

daschuer commented on src/library/trackmodel.h:
```diff
@@ -29,26 +29,27 @@ class TrackModel {
-        TRACKMODELCAPS_PURGE             = 0x10000,
-        TRACKMODELCAPS_REMOVE_PLAYLIST   = 0x20000,
-        TRACKMODELCAPS_REMOVE_CRATE      = 0x40000,
+        TRACKMODELCAPS_NONE              = 0x00000u,
```
Is this a clang-format complaint format. Even though it looks good, we should not do it, because the next edit here will mess up everything due to the pre-comit hook.

xerus2000 replied:
This will be changed in #2810 anyways

daschuer commented on src/library/trackset/crate/cratefeaturehelper.cpp:
```diff
@@ -21,8 +20,8 @@ QString CrateFeatureHelper::proposeNameForNewCrate(
             // Append suffix " 2", " 3", ...
-            proposedName = QString("%1 %2").arg(
-                    initialName, QString::number(suffixCounter));
+            proposedName = QString("%1 %2")
```
You may use  
initialName + QChar(' ') + QString::number(suffixCounter):

daschuer commented on src/library/trackset/crate/cratefeaturehelper.cpp:
```diff
@@ -87,19 +87,20 @@ CrateId CrateFeatureHelper::createEmptyCrate() {
-                    QString("%1 %2").arg(
-                            oldCrate.getName(), tr("copy" , "[noun]")));
+                    QString("%1 %2")
+                            .arg( oldCrate.getName(), tr("copy", "[noun]")));
```
The second argument in tr is deprecated. Use "//:"
You may also use + for string builder concatenation.

xerus2000 replied:
Fixed in d40ade3dd2 - I hope that's how you meant it, at least the tests still work ;)

Btw you can use Github change suggestions for such things, that way there is no ambiguity.

daschuer commented on src/library/trackset/crate/cratestorage.cpp:
```diff
@@ -357,13 +323,9 @@ CrateSelectResult CrateStorage::selectCratesByIds(
-                            CRATETABLE_NAME)));
+            mixxx::DbConnection::collateLexicographically(
+                    QString("SELECT * FROM %1 WHERE %2 %3 (%4) ORDER BY %5")
+                            .arg(CRATE_TABLE, CRATETABLE_ID, subselectPrefix, subselectForCrateIds, CRATETABLE_NAME)));
```
long line

xerus2000 replied:
This was intentional - my rationale was to only wrap the arguments is the format String is wrapped as well.

xerus2000 replied:
Anyways, fixed it, wrapped the format string as well

daschuer reviewed (changes requested):
There are some remaining style issues. 
I recommend to use the pre-commit hook, to avoid future style discussions.

xerus2000 commented:
> There are some remaining style issues.
> I recommend to use the pre-commit hook, to avoid future style discussions.

I USE CLANG-FORMAT! ALL THE TIME!

Sorry, but I can't take this anymore. For days almost all comments have been about issues of clang-format that I'm supposed to manually correct. I have better things to do. 
Please either merge my PR as is or refrain from any style comments until you have adjusted it to actually produce the style you want. I can also look into that, but so far my suggestions about config changes haven't been well received.

xeruf committed:
Reformat PlaylistDAO

xeruf committed:
Embrace QStringLiteral in TrackSet features

xeruf committed:
Replace tr "[noun]" with "//:"

Holzhaus commented on src/library/trackset/crate/cratefeature.cpp:
```diff
@@ -28,7 +28,7 @@ namespace {
         const CrateSummary& crateSummary) {
-    return QString("%1 (%2) %3")
+    return QStringLiteral("%1 (%2) %3")
             .arg(
```
Does QStringLiteral actually work with `.arg()`? I remember I tried that once and it didn't work...

uklotzde replied:
I can confirm that it does work and also remember that it didn't previously. Maybe switching to QStringBuilder did the trick?

xerus2000 replied:
The tests are all succeeding, and I hope there is at least _some_ testing covering the library. I changed this in so many places that it would be shocking if it wouldn't work and the tests would still succeed.

Holzhaus commented:
> > There are some remaining style issues.
> > I recommend to use the pre-commit hook, to avoid future style discussions.
> 
> I USE CLANG-FORMAT! ALL THE TIME!
> 
> Sorry, but I can't take this anymore. For days almost all comments have been about issues of clang-format that I'm supposed to manually correct. I have better things to do.
> Please either merge my PR as is or refrain from any style comments until you have adjusted it to actually produce the style you want. I can also look into that, but so far my suggestions about config changes haven't been well received.

`pre-commit` != `clang-format`. Please use the `pre-commit` hook, not just `clang-format`.
These hooks actually run `clang-format` twice: The first time it runs `clang-format`, the second time runs `clang-format` again with slightly different settings and only on long lines.

I didn't have any issues with formatting when using `pre-commit`. If you're are actually using that and it still doesn't format code correctly that's a bug and needs to be fixed.

xerus2000 commented:
> `pre-commit` != `clang-format`. Please use the `pre-commit` hook, not just `clang-format`.
> These hooks actually run `clang-format` twice: The first time it runs `clang-format`, the second time runs `clang-format` again with slightly different settings and only on long lines.

Thanks for the clarification! I assumed it would simply run clang-format.

Made that more clear on the wiki: https://www.mixxx.org/wiki/doku.php/coding_guidelines#pre-commit

xerus2000 commented:
I ran `pre-commit` over `src/library/library.cpp` and the whole `src/library` folder - nothing changed...

xeruf committed:
Fix comment spaces in src/library

uklotzde commented:
No need to delay this PR further due to code style discussions. We have identified some shortcomings of clang-format for multi-line string literals that could be resolved later.

daschuer committed:
clang-format

daschuer commented:
I am sorry to stress this formatting issue that much. 
But vertical scrolling during a review is a no go for me. 

I think the PR got to this state, because you probably have messed around with clang format during development. 
Luckily this reveals an issue with our CI clang-format test on Travis. 
I have added a PR that aims to fix it. 
https://github.com/mixxxdj/mixxx/pull/2816

Now you have three options to continue here: 
1. ```git rebase -i upstream/master``` 
    and amend each commit with the pre-commit hook changes. 
2. ```
git pull https://github.com/daschuer/mixxx.git pre-commit-clang
pre-commit run --source upstream/master --origin HEAD
git commit -a
git commit -a
```
3. Merge 
https://github.com/xerus2000/mixxx/pull/3

xerus2000 commented:
@daschuer I am leaning towards option 3, but this WILL create conflicts with my follow-up PR that I've already started working on because this took so long, which I am not fond of either...

Also, I am seeing quite a few things in there that make the code _less_ readable...

xerus2000 commented:
It's not as if I explicitly created new long lines. All the lines that got longer where solely because of clang-format. For the lines I worked on by hand I always checked the length.

I think the hard-wrap should leave a little buffer (maybe at 100), because sometimes a line needs to be a little longer to stay readable.

https://github.com/xerus2000/mixxx/pull/3#pullrequestreview-417269641
Here a few comments on wrongs of the formatter

xerus2000 commented:
> I think the PR got to this state, because you probably have messed around with clang format during development.

Nope, I didn't mess around. I have configured my IDE to automatically run clang-format on files I work on, if that doesn't produce the correct coding style without the pre-commit hook, it should be adjusted. There was only one option I tried, which was ` AlwaysBreakBeforeMultilineStrings: true` and didn't make much of a difference.

daschuer commented:
>  I am leaning towards option 3, but this WILL create conflicts with my follow-up PR that I've already started working on because this took so long, which I am not fond of either...

OK so go ahead. Since the clang-format commit does not change the code, the conflicts should be easy to solve.

xeruf committed:
Adjust reformattings

xerus2000 commented:
Alright, I hope we are finally done here :sweat_smile:

Holzhaus commented:
Some merge conflicts have emerged.

xerus2000 commented:
Me: Creates PR
Community: Reviews PR
Me: Let's merge
Community: First, let's fix the code style
Me: Okay, but let's finally merge!
Github: A wild conflict has appeared!
:sweat_smile:

xeruf committed:
Merge branch 'master' into refactor-basetrackset

xerus2000 commented:
Fixed, was merely about imports :)

xeruf committed:
Merge branch 'master' into refactor-basetrackset

xerus2000 commented:
@Holzhaus please merge, before more things appear :)

daschuer reviewed (approved):
LGTM, Thank you. And sorry for all the hassle.

Holzhaus reviewed (approved):
LGTM
